src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -241,11 +241,12 @@ def try_parse_tool_call_id_and_content(role_prompt):
 def try_parse_tool_calls(role_prompt):
     # customer can add ## in front of tool_calls for markdown highlight.
     # and we still support tool_calls without ## prefix for backward compatibility.
-    pattern = r"\n*#{0,2}\s*tool_calls\s*:\s*\n+\s*(\[.*?\])"
+    pattern = r"(\n*#{0,2}\s*tool_calls\s*:\s*\n+\s*)\["
```

Comment:
please kindly add comments to explain why must change to this pattern, which may help other maintainers in the future.

Reply:
Updated with comment